Kudu.Core/Jobs/BaseJobRunner.cs

```diff
@@ -140,6 +185,10 @@ private void CacheJobBinaries(JobBase job, IJobLogger logger)
                     UpdateAppConfigs(tempJobInstancePath);
 
                     _workingDirectory = tempJobInstancePath;
+
+                    // cache the file map snapshot for next time (to aid in detecting
+                    // file deletions)
+                    _cachedSourceDirectoryFileMap = sourceDirectoryFileMap;
```

Comment:
To close on this, we only do the directory diff when WorkingDirectory is not null, and whenever WorkingDirectory is not null, we have set _cachedSourceDirectoryFileMap to the initial set of source files we copied/cached. So we're good here. We're doing actual e2e tests to confirm this.